lightningd/peer_control.c

```diff
@@ -165,9 +165,8 @@ u32 feerate_min(struct lightningd *ld)
 	if (ld->config.ignore_fee_limits)
 		return 1;
 
-	/* Set this to average of slow and normal.*/
-	return (get_feerate(ld->topology, FEERATE_SLOW)
-		+ get_feerate(ld->topology, FEERATE_NORMAL)) / 2;
+	/* Set this to half of slow rate.*/
+	return get_feerate(ld->topology, FEERATE_SLOW) / 2;
```

Comment:
Could this potentially violate `feerate_floor()`? `get_feerate` gets its data from `topo->feerate`, which is loaded by `update_feerates`, which respects `feerate_floor()`.  The previous code averaged the slow and normal feerates.  Now we are getting half the slow feerate.  Could the result violate `feerate_floor` and is there some other point where we check against `feerate_floor()`? (it seems unlikely, as, `feerate_floor()` is `static` in `chaintopology`)

Reply:
Good point!  We should do that here, too.  I'll rework this logic.